ast/desugar/Desugar.cc

```diff
@@ -675,15 +759,17 @@ TreePtr node2TreeImpl(DesugarContext dctx, unique_ptr<parser::Node> what) {
                 auto arg = node2TreeImpl(dctx, std::move(andAsgn->right));
                 if (auto s = cast_tree<Send>(recv)) {
                     auto sendLoc = s->loc;
-                    auto [tempRecv, stats, readArgs, assgnArgs] = copyArgsForOpAsgn(dctx, s);
+                    auto [tempRecv, stats, numPosArgs, readArgs, assgnArgs] = copyArgsForOpAsgn(dctx, s);
+                    auto numPosAssgnArgs = numPosArgs + 1;
```

Comment:
The case that this is addressing looks like:
```
x = {}
x[10] &&= "Hello"
```
which will be desugared into:
```
x = {}
tmp = x[10]
if tmp
  x.[]=(10, "Hello")
else
  tmp
end
```
The send `x.[]=(10, "Hello")` needs to have a `numPosArgs` value of  2, where the original LHS had a `numPosArgs` value of 1.